### ISSUE TITLE
Errors in TypeScript 3.2.x

### ISSUE THREAD

**Issue by dehlak** [bazelbuild/rules_typescript#333]
Hey,

After updating typescript to version 3.2.x, our bazel build started failing with the following error:

```
external/build_bazel_rules_typescript/internal/tsc_wrapped/worker.ts(12,43): error TS2345: Argument of type '{}[]' is not assignable to parameter of type '[any?, ...any[]]'.
external/build_bazel_rules_typescript/internal/tsc_wrapped/worker.ts(20,32): error TS2345: Argument of type '{}[]' is not assignable to parameter of type '[any?, ...any[]]'.
```

Keeping TypeScript at version 3.1.x is obviously a simple workaround. Did anyone else experience the same issue?

Dehlak

**Comment by qzmfranklin** [comment]
This is already tracked by [this issue](https://github.com/Microsoft/TypeScript/issues/9358).

**Comment by alexeagle** [comment]
thanks @qzmfranklin

**Comment by Ubehebe** [comment]
What is the resolution of this issue? Should rules_typescript users just not upgrade to TypeScript 3.2? That doesn't sound great.

The errors reported by @dehlak are real errors when tsc_wrapped is compiled with TypeScript 3.2. Specifically, `--strict` mode now implies `--strictBindCallApply`, which improves type-checking for `apply` and causes compilation errors at the two call sites in worker.ts.

I sent [a PR](https://github.com/bazelbuild/rules_typescript/pull/358) in December that allowed tsc_wrapped to be compiled by both TypeScript 3.1 and 3.2, but it was closed in favor of [another PR](https://github.com/bazelbuild/rules_typescript/pull/348) that only fixed the 3.1 issues. I can reopen my PR if you'd like.
